lib/edition_tagger.rb

```diff
@@ -0,0 +1,27 @@
+
+
+class EditionTagger
+  def initialize(edition_tag_associations, logger)
+    @edition_tag_associations = edition_tag_associations
+    @logger = logger
+  end
+
+  def run
+    @edition_tag_associations.each do |association|
+      edition = Edition.where(slug: association[:slug], state: :published).first
+      association_tag_with_edition(edition, association[:tag])
+
+      edition.subsequent_siblings.where(state: :draft).each do |sibling|
+        association_tag_with_edition(sibling, association[:tag])
+      end
+    end
+  end
+
+  private
+
+  def association_tag_with_edition(edition, tag)
+    edition.browse_pages << tag
```

Comment:
It would be good to check the tag isn't already present here to avoid duplicates.